### PR TITLE
Order modules, read from /etc 

### PR DESCRIPTION
**Pull request by lnussel**

### CODE REVIEW

File: files/usr/share/jeos-firstboot/jeos-firstboot-functions [hunk 1]
```diff
@@ -54,6 +66,7 @@ call_module() {
 
 call_module_hook() {
         local hook="$1"
+	_init_modules
```
Comment (Vogtinator) [inline files/usr/share/jeos-firstboot/jeos-firstboot-functions:69]: Should be called in global scope on inclusion, like before.

File: files/usr/share/jeos-firstboot/jeos-firstboot-functions [hunk 2]
```diff
@@ -26,14 +26,26 @@ list=
...
+			fi
+		done
+	done
+}
+
+_init_modules() {
+	[ -z "$modules" ] || return 0
+	local module f
+	while read -r module f; do
+		[ -L "$f" ] && [ "$(readlink "$f")" = "/dev/null" ] && continue
```
Comment (Vogtinator) [inline files/usr/share/jeos-firstboot/jeos-firstboot-functions:44]: IMO this needs to be a full if. The `&&` shorthand syntax should only be used once per line.

Comment (lnussel) [reply files/usr/share/jeos-firstboot/jeos-firstboot-functions]: :shrug:

File: files/usr/share/jeos-firstboot/jeos-firstboot-functions (hunk 1 above)
Comment (lnussel) [reply files/usr/share/jeos-firstboot/jeos-firstboot-functions]: moved it to after dryrun etc in main file. doing that on source seems ugly to me

**Comment by lnussel** [comment]
ping?

File: files/usr/lib/systemd/system/jeos-firstboot.service [hunk 3]
```diff
@@ -34,6 +34,7 @@ ExecStart=/usr/sbin/jeos-firstboot
 StandardOutput=tty
 StandardInput=tty
 #StandardError=tty
+KeyringMode=shared
```
Comment (Vogtinator) [inline files/usr/lib/systemd/system/jeos-firstboot.service:37]: Needs an explanation.

File: files/usr/sbin/jeos-firstboot [hunk 4]
```diff
@@ -4,14 +4,22 @@
 
 set -e
 
+# set to --keep-tite to use altenate screen. Better for debugging but causes flicker
```
Comment (Vogtinator) [inline files/usr/sbin/jeos-firstboot:7]:
```suggestion
# set to --keep-title to use alternate screen. Better for debugging but causes flicker
```

File: files/usr/sbin/jeos-firstboot [hunk 5]
```diff
@@ -4,14 +4,22 @@
...
+dialog_altenate_screen=
```
Comment (Vogtinator) [inline files/usr/sbin/jeos-firstboot:8]: Typo. This is used in jeos-firstboot-functions and needs to be defined there. Currently it's undefined when running jeos-config.

**Comment by Vogtinator** [comment]
I'll rebase #97. Can you then rebase this on that PR instead?

**Review by Vogtinator — CHANGES_REQUESTED** [review]

**Comment by lnussel** [comment]
if you add 97 that would screw up mine I guess? Would be unfair after a year letting that other thing linger

File: files/usr/share/jeos-firstboot/jeos-firstboot-functions [hunk 6]
```diff
@@ -110,11 +124,18 @@ findlocales()
         list=()
         local l locale
         # List only locales which are both in live-langset-data and glibc-locale(-base)
-        for l in /usr/share/langset/*; do
-                locale="${l#/usr/share/langset/}"
-                [ -d "/usr/lib/locale/${locale}.utf8" ] || continue
-                list+=("${locale}" '')
-        done
+        if [ -d /usr/share/langset ]; then
```
Comment (Vogtinator) [inline files/usr/share/jeos-firstboot/jeos-firstboot-functions:127]: jeos-firstboot calls langset.sh so this is a hard requirement

**Comment by Vogtinator** [comment]
> if you add 97 that would screw up mine I guess?

I thought I wrote in here that this duplicates some parts of #97, but I can't find that comment. Maybe I never submitted it...

>  Would be unfair after a year letting that other thing linger

IMO this PR is a reason to finish up #97, while previously it was just "nice to have".

File: files/usr/sbin/jeos-firstboot (hunk 4 above)
Comment (lnussel) [reply files/usr/sbin/jeos-firstboot]: it's actually called --keep-tite

File: files/usr/share/jeos-firstboot/jeos-firstboot-functions (hunk 6 above)
Comment (lnussel) [reply files/usr/share/jeos-firstboot/jeos-firstboot-functions]: yeah, was good enough for dry run though

**Comment by lnussel** [comment]
well, you can rebase on this one if you have to rebase anyway :)

**Comment by Vogtinator** [comment]
> well, you can rebase on this one if you have to rebase anyway :)

"Fix calling without installation" is incomplete and I personally don't like `type -P $0` or using the `${pfx%/*/*}` syntax. At that point it doesn't matter much whether #97 is changed to be more like the commit here or the other way around. I'll just merge them both together and add a `Co-authored-by` tag. Does that sound good?

**Comment by lnussel** [comment]
fine with me. calling dirname etc requires external programs for something bash can do by itself. bad style :)

**Comment by lnussel** [comment]
rebased on yours

**Comment by lnussel** [comment]
ok now?

File: files/usr/share/jeos-firstboot/jeos-firstboot-functions [hunk 7]
```diff
@@ -27,15 +27,28 @@ list=
...
+}
+
+init_modules() {
+	[ -z "$modules" ] || return 0
+	local module f
+	while read -r module f; do
+		if [ -L "$f" ] && [ "$(readlink "$f")" = "/dev/null" ]; then
+			continue
+		fi
+		. "$f"
```
Comment (Vogtinator) [inline files/usr/share/jeos-firstboot/jeos-firstboot-functions:48]: Needs to do `source "$f"` within an if to not kill the entire process if a module has a syntax error

**Comment by lnussel** [comment]
updated

**Review by Vogtinator — APPROVED** [review]

**Pull request merged**